src/api/controllers/BaseController.ts

```diff
@@ -0,0 +1,40 @@
+import { EventEmitter } from 'eventemitter3';
+
+export interface BaseEvents
+{
+    hidden: ()=> void;
+    shown: ()=> void;
+    opened: ()=> void;
+    closed: ()=> void;
+}
+
+export interface BaseControllerOptions
+{
+    style?: any;
+    view?: any;
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```

Comment:
Shouldn't have these unless there is a third party fault

Reply:
react wont compile if there are type errors

Reply:
Can you not fix the type error?